### PR TITLE
ARROW-3163: [Python] Add missing Cython dependency to source package

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -514,22 +514,28 @@ def has_ext_modules(foo):
...
     'six >= 1.0.0',
     'futures; python_version < "3.2"'
-)
+]
+
+setup_requires = []
 
+cython_require = 'cython >= 0.27'
+
+if {'bdist_wheel'}.intersection(sys.argv):
```
Comment (kou) [inline python/setup.py]: I'm sorry but I don't know about this approach is good...

**Review by xhochy — CHANGES_REQUESTED** [review]
No, the current `as-is` setup is exactly what's needed. Please do not merge. `setup_requires` is exactly the place where one needs to specify build-time requirements.

The proper solution to your problem is adding a  `pyproject.toml` https://www.python.org/dev/peps/pep-0518/ to specify build-time requirements before `setup.py` is executed.

**Comment by pitrou** [comment]
Agreed with @xhochy. There doesn't seem to be a reason for this PR.

**Comment by kou** [comment]
Thanks. I didn't know PEP 518.
I've added `pyproject.toml` and confirmed that it works well with the latest pip.

By the way, can we remove `setup_requires` from `setup.py`? Or should we keep it for old distutils (pip?)?

**Review by xhochy — APPROVED** [review]
+1, LGTM

We should keep `setup_requires` around for a while.

**Pull request closed without merging**